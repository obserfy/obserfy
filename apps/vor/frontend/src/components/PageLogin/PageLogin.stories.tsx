import { FC } from "react"
import PageLogin from "./PageLogin"

export default {
  title: "Page/PageLogin",
  component: PageLogin,
  parameters: {
    componentSubtitle: "Just a simple PageLogin",
  },
}

export const Basic: FC = () => <PageLogin />
