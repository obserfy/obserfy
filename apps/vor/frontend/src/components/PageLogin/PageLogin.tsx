import { FC, useState } from "react"
import { Box, Button, Flex } from "theme-ui"
import { captureException } from "../../analytics"
import Input from "../Input/Input"
import { Link } from "../Link/Link"
import { Typography } from "../Typography/Typography"
import Icon from "../Icon/Icon"
import { ReactComponent as InfoIcon } from "../../icons/info.svg"
import BrandBanner from "../BrandBanner/BrandBanner"
import usePostLogin from "../../hooks/api/usePostLogin"
import LoadingIndicator from "../LoadingIndicator/LoadingIndicator"

export const PageLogin: FC = () => {
  const [email, setEmail] = useState("")
  const [password, setPassword] = useState("")
  const { mutateAsync, error, isLoading } = usePostLogin()

  return (
    <Box>
      <BrandBanner />

      <Flex sx={{ justifyContent: "center" }}>
        <Box
          as="form"
          px={3}
          sx={{ width: "100%", maxWidth: "maxWidth.xsm" }}
          onSubmit={async (event) => {
            try {
              event.preventDefault()
              await mutateAsync({ email, password })
            } catch (e) {
              captureException(e)
            }
          }}
        >
          <Typography.H5 sx={{ fontWeight: "bold" }} my={3}>
            Login
          </Typography.H5>
          <Input
            sx={{ width: "100%" }}
            label="Email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
            required
            mb={2}
          />
          <Input
            type="password"
            sx={{ width: "100%" }}
            label="Password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
            required
          />

          <Button my={3} sx={{ width: "100%" }}>
            {isLoading && <LoadingIndicator />}
            Login
          </Button>
          <Typography.Body
            my={3}
            color="danger"
            sx={{
              textAlign: "center",
              fontSize: 1,
              width: "100%",
              fontWeight: "bold",
            }}
          >
            {error ? "Wrong email or password" : ""}
          </Typography.Body>
          <Typography.Body
            color="textMediumEmphasis"
            sx={{ textAlign: "center" }}
          >
            <Link to="/forgot-password" sx={{ color: "textPrimary" }}>
              Forgot password
            </Link>
          </Typography.Body>
          <Typography.Body
            mt={5}
            color="textMediumEmphasis"
            sx={{ textAlign: "center" }}
          >
            Don&apos;t have an account?{" "}
            <Link to="/register" sx={{ color: "textPrimary" }}>
              Sign Up
            </Link>
          </Typography.Body>
        </Box>
      </Flex>
      <Box mx="auto" p={3} sx={{ maxWidth: "maxWidth.xsm" }}>
        <Flex
          p={3}
          backgroundColor="primaryLighter"
          sx={{
            borderRadius: "default",
            alignItems: "center",
          }}
        >
          <Icon as={InfoIcon} fill="primaryDark" />
          <Typography.Body sx={{ fontSize: 1, lineHeight: 1.4 }} ml={2}>
            Are you a parent?{" "}
            <a
              href="https://parent.obserfy.com/api/auth/login"
              sx={{ color: "textPrimary" }}
            >
              Go to parent dashboard
            </a>
          </Typography.Body>
        </Flex>
      </Box>
    </Box>
  )
}

export default PageLogin
