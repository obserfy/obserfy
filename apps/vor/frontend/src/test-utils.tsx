/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
