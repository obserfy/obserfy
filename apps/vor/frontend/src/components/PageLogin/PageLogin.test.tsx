import { render } from "../../test-utils"
import { Basic } from "./PageLogin.stories"

describe("PageLogin", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
