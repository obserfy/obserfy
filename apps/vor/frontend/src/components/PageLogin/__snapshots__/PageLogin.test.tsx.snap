// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLogin should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  padding: 16px;
  width: 100%;
  max-width: 500px;
}

.emotion-3 {
  display: inline-block;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-left: 8px;
  font-size: 20px;
  font-weight: bold;
  line-height: 1.2;
}

.emotion-6 {
  font-weight: normal;
  white-space: nowrap;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 16px;
  padding-right: 16px;
  width: 100%;
  max-width: 500px;
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: bold;
  line-height: heading;
  color: var(--theme-ui-colors-text);
  text-rendering: optimizeLegibility;
  font-size: 24px;
  margin-top: 16px;
  margin-bottom: 16px;
  font-weight: bold;
}

.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: var(--theme-ui-colors-textMediumEmphasis);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-11 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-bottom: 4px;
}

.emotion-12 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-13 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text);
  margin-bottom: 8px;
  --theme-ui-input-autofill-bg: var(--theme-ui-colors-background);
  width: 100%;
  width: 100%;
}

.emotion-13:autofill,
.emotion-13:autofill:hover,
.emotion-13:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-13:autofill:first-line,
.emotion-13:autofill:hover:first-line,
.emotion-13:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-13:-webkit-autofill,
.emotion-13:-webkit-autofill:hover,
.emotion-13:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-13:-webkit-autofill:first-line,
.emotion-13:-webkit-autofill:hover:first-line,
.emotion-13:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-13:hover,
.emotion-13:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-13::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-13::-moz-placeholder {
  font-style: italic;
}

.emotion-13:-ms-input-placeholder {
  font-style: italic;
}

.emotion-13::placeholder {
  font-style: italic;
}

.emotion-13:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border);
  color: var(--theme-ui-colors-textDisabled);
}

.emotion-17 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text);
  --theme-ui-input-autofill-bg: var(--theme-ui-colors-background);
  width: 100%;
  width: 100%;
}

.emotion-17:autofill,
.emotion-17:autofill:hover,
.emotion-17:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-17:autofill:first-line,
.emotion-17:autofill:hover:first-line,
.emotion-17:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-17:-webkit-autofill,
.emotion-17:-webkit-autofill:hover,
.emotion-17:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-17:-webkit-autofill:first-line,
.emotion-17:-webkit-autofill:hover:first-line,
.emotion-17:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-17:hover,
.emotion-17:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-17::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-17::-moz-placeholder {
  font-style: italic;
}

.emotion-17:-ms-input-placeholder {
  font-style: italic;
}

.emotion-17::placeholder {
  font-style: italic;
}

.emotion-17:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border);
  color: var(--theme-ui-colors-textDisabled);
}

.emotion-18 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-onPrimary);
  background-color: var(--theme-ui-colors-primary);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: rgba(27, 31, 35, 0.1) 0px 1px 0px 0px,rgba(255, 255, 255, 0.03) 0px 1px 0px 0px inset;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  margin-top: 16px;
  margin-bottom: 16px;
  width: 100%;
}

.emotion-18:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-18:hover,
.emotion-18:focus {
  background-color: var(--theme-ui-colors-primaryDark);
}

.emotion-18:hover:disabled,
.emotion-18:focus:disabled {
  background-color: var(--theme-ui-colors-primary);
}

.emotion-19 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-top: 16px;
  margin-bottom: 16px;
  color: var(--theme-ui-colors-danger);
  text-align: center;
  font-size: 14px;
  width: 100%;
  font-weight: bold;
}

.emotion-20 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  color: var(--theme-ui-colors-textMediumEmphasis);
  text-align: center;
}

.emotion-21 {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-22 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-top: 64px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  text-align: center;
}

.emotion-24 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  padding: 16px;
  max-width: 500px;
}

.emotion-25 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
  background-color: var(--theme-ui-colors-primaryLighter);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 6px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-26 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-primaryDark);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

.emotion-27 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-left: 8px;
  font-size: 14px;
  line-height: 1.4;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <a
          class="emotion-3"
          href="https://obserfy.com"
        >
          <div
            class="emotion-4"
          >
            <p
              class="emotion-5"
            >
              Obserfy
               
              <span
                class="emotion-6"
              >
                for Teachers
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="emotion-7"
      >
        <form
          class="emotion-8"
        >
          <h5
            class="emotion-9"
          >
            Login
          </h5>
          <label
            class="emotion-10"
          >
            <div
              class="emotion-11"
            >
              Email
            </div>
            <div
              class="emotion-12"
            >
              <input
                class="emotion-13"
                required=""
                value=""
              />
            </div>
          </label>
          <label
            class="emotion-10"
          >
            <div
              class="emotion-11"
            >
              Password
            </div>
            <div
              class="emotion-12"
            >
              <input
                class="emotion-17"
                required=""
                type="password"
                value=""
              />
            </div>
          </label>
          <button
            class="emotion-18"
          >
            Login
          </button>
          <p
            class="emotion-19"
          />
          <p
            class="emotion-20"
          >
            <a
              class="emotion-21"
              href="/forgot-password"
            >
              Forgot password
            </a>
          </p>
          <p
            class="emotion-22"
          >
            Don't have an account?
             
            <a
              class="emotion-21"
              href="/register"
            >
              Sign Up
            </a>
          </p>
        </form>
      </div>
      <div
        class="emotion-24"
      >
        <div
          class="emotion-25"
        >
          <div
            class="emotion-26"
          />
          <p
            class="emotion-27"
          >
            Are you a parent?
             
            <a
              class="emotion-21"
              href="https://parent.obserfy.com/api/auth/login"
            >
              Go to parent dashboard
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
